File: src/main/java/ar/edu/unq/epers/unidad5/service/ProductoServiceImpl.java
package ar.edu.unq.epers.unidad5.service;

import ar.edu.unq.epers.unidad5.dao.ProductoDAO;
import ar.edu.unq.epers.unidad5.model.Producto;
import ar.edu.unq.epers.unidad5.model.PrecioPromedio;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductoServiceImpl implements ProductoService {

    private final ProductoDAO productoDAO;

    public ProductoServiceImpl(ProductoDAO productoDAO) {
        this.productoDAO = productoDAO;
    }

    @Override
    public Producto save(Producto producto) {
        return productoDAO.save(producto);
    }

    @Override
    public void saveAll(List<Producto> productos) {
        productoDAO.saveAll(productos);
    }

    @Override
    public List<Producto> findByBrand(String marca) {
        return productoDAO.findByBrand(marca);
    }

    @Override
    public Producto getByCode(String code) {
        return productoDAO.getByCode(code);
    }

    @Override
    public List<Producto> findByPrice(int precio) {
        return productoDAO.findByPrice(precio);
    }

    @Override
    public List<Producto> findByPriceRange(int min, int max) {
        return productoDAO.findByPriceRange(min, max);
    }

    @Override
    public List<PrecioPromedio> getPrecioPromedio(List<String> codigos) {
        return productoDAO.getPrecioPromedio(codigos);
    }

    @Override
    public void deleteAll() {
        productoDAO.deleteAll();
    }
}

File: src/main/kotlin/ar/edu/unq/epers/unidad5/service/ProductoServiceImpl.kt
package ar.edu.unq.epers.unidad5.service

import ar.edu.unq.epers.unidad5.dao.ProductoDAO
import ar.edu.unq.epers.unidad5.model.PrecioPromedio
import ar.edu.unq.epers.unidad5.model.Producto
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ProductoServiceImpl: ProductoService {

    @Autowired
    lateinit var productoDAO: ProductoDAO

    override fun save(producto: Producto): Producto {
        return productoDAO.save(producto)
    }

    override fun saveAll(productos: List<Producto>) {
        productoDAO.saveAll(productos)
    }

    override fun findByBrand(marca: String): List<Producto> {
        return productoDAO.findByBrand(marca)
    }

    override fun getByCode(code: String): Producto {
        return productoDAO.getByCode(code)
    }

    override fun findByPrice(precio: Int): List<Producto> {
        return productoDAO.findByPrice(precio)
    }

    override fun findByPriceRange(min: Int, max: Int): List<Producto> {
        return productoDAO.findByPriceRange(min, max)
    }

    override fun getPrecioPromedio(codigos: List<String>): List<PrecioPromedio> {
        return productoDAO.getPrecioPromedio(codigos)
    }

    override fun deleteAll() {
        productoDAO.deleteAll()
    }

}